central/auth/m2m/set.go

```diff
@@ -24,9 +24,10 @@ var (
 //go:generate mockgen-wrapper
 type TokenExchangerSet interface {
 	NewTokenExchangerFromConfig(ctx context.Context, config *storage.AuthMachineToMachineConfig) (TokenExchanger, error)
-	UpsertTokenExchanger(exchanger TokenExchanger, issuer string) error
+	UpsertTokenExchanger(exchanger TokenExchanger, issuer string)
 	RemoveTokenExchanger(issuer string) error
 	GetTokenExchanger(issuer string) (TokenExchanger, bool)
+	RollbackExchanger(issuer string) error
```

Comment:
It seems that our lives could've been simpler if `issuerFactory` allowed us to create an issuer without registering, or to do an issuer modification. Perhaps something to think about wrt future refactorings